### PR TITLE
Add script to check User disallow permissions

### PR DESCRIPTION
**Pull request by OVikhrov**
@maria-mih , please review

### CODE REVIEW

File: files/PTU_with_permissions_for_app_1234567.json [hunk 1]
```diff
@@ -0,0 +1,2073 @@
+{
```
Comment (maria-mih) [inline files/PTU_with_permissions_for_app_1234567.json]: Review started

File: test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+-- 1. assign only the permissions for which there is user consent in the local policy table and permissions (if any) that are listed in 'pre_DataConsent' section (that is, 'default_hmi', 'groups' and other) for this application 
+-- 2. notify HMI via onPermissionsChange() notification about the applied permissions.
+-- 3. notify mobile app via onPermissionsChange() notification about the applied permissions.
+-- Preconditions:
+-- 1. Delete policy.sqlite file, app_info.dat files
+-- 2. <Device> is connected to SDL and consented by the User, <App> is running on that device.
+-- 3. <App> is registered with SDL and is present in HMI list of registered aps.
+-- 4. Local PT has permissions for <App> that require User`s consent.
+--
+-- Requirement summary:
```
Comment (maria-mih) [inline test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua]: This section should be first please move it to top. 
Also please add here only name of requirement. In your case it should look like:
-- Requirement summary:
-- [Policies]: User-consent "NO"

File: test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+-- On getting user consent information from HMI and storage into LocalPolicyTable ,
```
Comment (maria-mih) [inline test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua]: This should be part of description

File: test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua [hunk 4]
```diff
@@ -0,0 +1,309 @@
+---------------------------------------------------------------------------------------------
+-- Description:
+-- On getting negative user consent from HMI and storage this data into LocalPolicyTable, SDL must:
+-- 1. assign only the permissions for which there is user consent in the local policy table and permissions (if any) that are listed in 'pre_DataConsent' section (that is, 'default_hmi', 'groups' and other) for this application 
+-- 2. notify HMI via onPermissionsChange() notification about the applied permissions.
+-- 3. notify mobile app via onPermissionsChange() notification about the applied permissions.
+-- Preconditions:
```
Comment (maria-mih) [inline test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua]: Please check template how sections are organized. https://github.com/CustomSDL/sdl_atf_test_scripts/pull/32 
Description:
Put here the short description
1. Used preconditions 
Put here the preconditions
2. Performed steps 
Only here list what actions are done 
Expected result:
Put here what are the expectations

File: test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua [hunk 5]
```diff
@@ -0,0 +1,309 @@
...
+      {hmi_levels = {"BACKGROUND", "FULL", "LIMITED", "NONE"}},
+  }}
+  data = json.encode(data)
+  file = io.open(pathToFile, "w")
+  file:write(data)
+  file:close()
+end
+
+--[[ Precondition ]]
+local function Precondition()
```
Comment (maria-mih) [inline test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua]: Please check template how this section should be introduced and organized

File: test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua [hunk 6]
```diff
@@ -0,0 +1,309 @@
...
+
+  commonFunctions:userPrint(34, "-- Precondition --")
+
+  function Test:StopSDL()
```
Comment (maria-mih) [inline test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua]: Please add to names of all functions here prefix Precondition_
in this case 
function Test:Precondition_StopSDL()

File: test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua [hunk 7]
```diff
@@ -0,0 +1,309 @@
...
+    StopSDL()
+  end
+
+  --ToDo: shall be removed when issue: "SDL doesn't stop at execution ATF function StopSDL()" is fixed
+  function Test:SDLForceStop()
```
Comment (maria-mih) [inline test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua]: Please check comment about name above

File: test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua [hunk 8]
```diff
@@ -0,0 +1,309 @@
...
+    commonFunctions:SDLForceStop()
+  end
+
+  function Test:DeleteLogsAndPolicyTable()
```
Comment (maria-mih) [inline test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua]: Please check comment about name above

File: test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua [hunk 9]
```diff
@@ -0,0 +1,309 @@
...
+    commonSteps:DeleteLogsFiles()
+    commonSteps:DeletePolicyTable()
+  end
+
+  function Test:Backup_sdl_preloaded_pt()
```
Comment (maria-mih) [inline test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua]: Please check comment about name above

File: test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua [hunk 10]
```diff
@@ -0,0 +1,309 @@
...
+    BackupPreloaded()
+  end
+
+  function Test:SetPermissionsForPre_DataConsent()
```
Comment (maria-mih) [inline test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua]: Please check comment about name above

File: test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua [hunk 11]
```diff
@@ -0,0 +1,309 @@
...
+    SetPermissionsForPre_DataConsent()
+  end
+
+  function Test:StartSDL_FirstLifeCycle()
```
Comment (maria-mih) [inline test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua]: Please check comment about name above

File: test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua [hunk 12]
```diff
@@ -0,0 +1,309 @@
...
+    StartSDL(config.pathToSDL, config.ExitOnCrash)
+  end
+
+  function Test:InitHMI_FirstLifeCycle()
```
Comment (maria-mih) [inline test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua]: Please check comment about name above

File: test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua [hunk 13]
```diff
@@ -0,0 +1,309 @@
...
+    self:initHMI()
+  end
+
+  function Test:InitHMI_onReady_FirstLifeCycle()
```
Comment (maria-mih) [inline test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua]: Please check comment about name above

File: test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua [hunk 14]
```diff
@@ -0,0 +1,309 @@
...
+    self:initHMI_onReady()
+  end
+
+  function Test:ConnectMobile_FirstLifeCycle()
```
Comment (maria-mih) [inline test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua]: Please check comment about name above

File: test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua [hunk 15]
```diff
@@ -0,0 +1,309 @@
...
+    self:connectMobile()
+  end
+
+  function Test:StartSession()
```
Comment (maria-mih) [inline test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua]: Please check comment about name above

File: test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua [hunk 16]
```diff
@@ -0,0 +1,309 @@
...
+    self.mobileSession = mobile_session.MobileSession(self, self.mobileConnection)
+    self.mobileSession:StartService(7)
+  end
+
+  function Test:RestorePreloadedPT()
```
Comment (maria-mih) [inline test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua]: Please check comment about name above

File: test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua [hunk 17]
```diff
@@ -0,0 +1,309 @@
...
+    RestorePreloadedPT()
+  end
+
+  function Test:Activate_App_And_Consent_Device()
```
Comment (maria-mih) [inline test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua]: Please check comment about name above

File: test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua [hunk 18]
```diff
@@ -0,0 +1,309 @@
...
+                :Do(function(_,data)
+                    self.hmiConnection:SendResponse(data.id,"BasicCommunication.ActivateApp", "SUCCESS", {})
+                    EXPECT_NOTIFICATION("OnHMIStatus", {hmiLevel = "FULL", systemContext = "MAIN"})
+                  end)
+              end)
+          end)
+      end)
+  end
+
+  function Test:DeactivateApp()
```
Comment (maria-mih) [inline test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua]: Please check comment about name above

File: test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua [hunk 19]
```diff
@@ -0,0 +1,309 @@
...
+                end
+                RUN_AFTER(to_run, 800)
+                self.mobileSession:ExpectResponse(CorIdSystemRequest, {success = true, resultCode = "SUCCESS"})
+              end)
+          end)
+      end)
+  end
+
+end
+Precondition()
```
Comment (maria-mih) [inline test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua]: Please remove this line it is redundant

File: test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua [hunk 20]
```diff
@@ -0,0 +1,309 @@
...
+
+--[[ Test ]]
+function Test:Step1_User_Disallow_New_Permissions_After_App_Activation()
```
Comment (maria-mih) [inline test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua]: Please check in template https://github.com/CustomSDL/sdl_atf_test_scripts/pull/32  how section Test is organized.
Also add to name of all functions in this section TestStep and remove the numbers.

File: test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua [hunk 21]
```diff
@@ -0,0 +1,309 @@
...
+
+  commonFunctions:userPrint(34, "-- Test --")
```
Comment (maria-mih) [inline test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua]: Remove, this is redundant if section organized according to template

File: test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua [hunk 22]
```diff
@@ -0,0 +1,309 @@
...
+      local functionalGroupID = data.result.allowedFunctions[1].id
+      self.hmiConnection:SendNotification("SDL.OnAppPermissionConsent",
+        { appID = self.applications["SPT"], source = "GUI", consentedFunctions = {{name = "New_permissions", allowed = false, id = functionalGroupID} }})
+    end)
+  EXPECT_NOTIFICATION("OnPermissionsChange", {})
+  -- Permissions based on preloaded_pt file
+  --ToDo: (vikhrov) develop functions for extracting permissions for specific groups and compare it with permissionItem from OnPermissionsChange
+end
+
+function Test:Step2_Check_RPC_Disallowed_By_User()
```
Comment (maria-mih) [inline test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua]: Check comment above about name of functions

File: test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua [hunk 23]
```diff
@@ -0,0 +1,309 @@
...
+      mediaClock = "00:00:01",
+      mainField1 = "Show1"
+    })
+  EXPECT_HMICALL("UI.Show", {}):Do(function(_,data)
+      self.hmiConnection:SendResponse(data.id, "UI.Show", "SUCCESS", { })
+    end)
+  EXPECT_RESPONSE(CorIdRAI, { success = true, resultCode = "SUCCESS"})
+end
+
+function Test:Step3_Check_Disallowed_RPC()
```
Comment (maria-mih) [inline test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua]: Check comment above about name of functions

File: test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua [hunk 24]
```diff
@@ -0,0 +1,309 @@
...
+  local cid = self.mobileSession:SendRPC("AddSubMenu",
+    {
+      menuID = 1000,
+      position = 500,
+      menuName ="SubMenupositive"
+    })
+  EXPECT_RESPONSE(cid, { success = false, resultCode = "DISALLOWED" })
+end
+
+--[[ Postcondition ]]
```
Comment (maria-mih) [inline test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua]: Please organize section according to template https://github.com/CustomSDL/sdl_atf_test_scripts/pull/32

File: test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua (hunk 2 above)
Comment (OVikhrov) [inline test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua]: Accepted. Corrected in fe816a6

File: test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua (hunk 3 above)
Comment (OVikhrov) [inline test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua]: Accepted. Corrected in fe816a6

File: test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua (hunk 6 above)
Comment (OVikhrov) [inline test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua]: Accepted. Corrected in fe816a6

File: test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua (hunk 7 above)
Comment (OVikhrov) [inline test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua]: Accepted. Corrected in fe816a6

File: test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua (hunk 8 above)
Comment (OVikhrov) [inline test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua]: Accepted. Corrected in fe816a6

File: test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua (hunk 9 above)
Comment (OVikhrov) [inline test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua]: Accepted. Corrected in fe816a6

File: test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua (hunk 12 above)
Comment (OVikhrov) [inline test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua]: Accepted. Corrected in fe816a6

File: test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua (hunk 13 above)
Comment (OVikhrov) [inline test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua]: Accepted. Corrected in fe816a6

File: test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua (hunk 14 above)
Comment (OVikhrov) [inline test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua]: Accepted. Corrected in fe816a6

File: test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua (hunk 17 above)
Comment (OVikhrov) [inline test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua]: Accepted. Corrected in fe816a6

File: test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua (hunk 18 above)
Comment (OVikhrov) [inline test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua]: Accepted. Corrected in fe816a6

File: test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua (hunk 19 above)
Comment (OVikhrov) [inline test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua]: Accepted. Corrected in fe816a6

File: test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua (hunk 20 above)
Comment (OVikhrov) [inline test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua]: Accepted. Corrected in fe816a6

File: test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua (hunk 21 above)
Comment (OVikhrov) [inline test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua]: Accepted. Corrected in fe816a6

File: test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua (hunk 22 above)
Comment (OVikhrov) [inline test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua]: Accepted. Corrected in fe816a6

File: test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua (hunk 23 above)
Comment (OVikhrov) [inline test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua]: Accepted. Corrected in fe816a6

File: test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua (hunk 24 above)
Comment (OVikhrov) [inline test_scripts/Polices/App_Permissions/ATF_HP_User_Consent_NO.lua]: Accepted. Corrected in fe816a6

**Comment by OVikhrov** [comment]
@maria-mih Please, review next iteration

**Comment by istoimenova** [comment]
@LuxoftAKutsan please merge.

**Pull request merged**